lib/IRGen/IRGenSIL.cpp

```diff
@@ -6698,8 +6698,12 @@ void IRGenSILFunction::visitConvertFunctionInst(swift::ConvertFunctionInst *i) {
       fnType->getRepresentation() != SILFunctionType::Representation::Block) {
     auto *fn = temp.claimNext();
     Explosion res;
-    auto sig = IGM.getSignature(fnType);
-    res.add(Builder.CreateBitCast(fn, sig.getType()->getPointerTo()));
+    auto &fnTI = IGM.getTypeInfoForLowered(fnType);
```

Comment:
Can we guard this so this only happens for autoDiff functions?

Reply:
There is nothing autodiff specific here.